app/models/miq_server/log_management.rb

```diff
@@ -124,13 +124,14 @@ def last_log_sync_message
   def post_logs(options)
     taskid = options[:taskid]
     task = MiqTask.find(taskid)
+    context = options[:context]
 
     # the current queue item and task must be errored out on exceptions so re-raise any caught errors
-    raise "Log depot settings not configured" unless log_depot
-    log_depot.update_attributes(:support_case => options[:support_case].presence)
+    raise "Log depot settings not configured" unless log_depot(context)
+    log_depot(context).update_attributes(:support_case => options[:support_case].presence)
 
-    post_historical_logs(taskid) unless options[:only_current]
-    post_current_logs(taskid)
+    post_historical_logs(taskid, context) unless options[:only_current]
+    post_current_logs(taskid, context)
```

Comment:
Let's pass the call `log_depot(context)` to get the depot and pass it into these methods... we don't use the context for anything else in those methods... at least it doesn't look like it.